tests/src/tests/test_utils.go

```diff
@@ -0,0 +1,324 @@
+
+func LogMessage(message string) string {
+	testLogger.Println(message)
+	return message
+}
+
+type TestDevfile struct {
+	SchemaDevFile   schema.Devfile
+	FileName        string
+	ParsedSchemaObj parser.DevfileObj
+	SchemaParsed    bool
+}
+
+var StringCount int = 0
+
+var RndSeed int64 = time.Now().UnixNano()
+
+func GetRandomUniqueString(n int, lower bool) string {
+	StringCount++
+	return fmt.Sprintf("%s%04d", GetRandomString(n, lower), StringCount)
+}
+
+func setRandSeed() {
+	RndSeed++
+	rand.Seed(RndSeed)
+}
+
+const schemaBytes = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+
+func GetRandomString(n int, lower bool) string {
+	setRandSeed()
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = schemaBytes[rand.Intn(len(schemaBytes)-1)]
+	}
+	randomString := string(b)
+	if lower {
+		randomString = strings.ToLower(randomString)
+	}
+	return randomString
+}
+
+func GetGroupKinds() []schema.CommandGroupKind {
+	setRandSeed()
+	return []schema.CommandGroupKind{schema.BuildCommandGroupKind, schema.RunCommandGroupKind, schema.TestCommandGroupKind, schema.DebugCommandGroupKind}
+}
+
+func GetRandomGroupKind() schema.CommandGroupKind {
+	setRandSeed()
+	return GetGroupKinds()[rand.Intn(len(GetGroupKinds()))]
+}
+
+func GetBinaryDecision() bool {
+	return GetRandomDecision(1, 1)
+}
+
+func GetRandomDecision(success int, failure int) bool {
+	setRandSeed()
+	return rand.Intn(success+failure) > failure-1
+}
+
+func GetRandomNumber(max int) int {
+	setRandSeed()
+	return rand.Intn(max) + 1
+}
+
+func GetDevfile(fileName string) TestDevfile {
+	testDevfile := TestDevfile{}
+	testDevfile.SchemaDevFile = schema.Devfile{}
+	testDevfile.FileName = fileName
+	testDevfile.SchemaDevFile.SchemaVersion = "2.0.0"
+	testDevfile.SchemaParsed = false
+	return testDevfile
+}
+
+func (devfile *TestDevfile) CreateDevfile(useParser bool) error {
+	var err error
+
+	fileName := devfile.FileName
+	if !strings.HasSuffix(fileName, ".yaml") {
+		fileName += ".yaml"
+	}
+
+	if useParser {
+		LogMessage(fmt.Sprintf("   .......use Parser to write devfile %s", fileName))
+		newDevfile, err := devfileData.NewDevfileData(devfile.SchemaDevFile.SchemaVersion)
+		if err != nil {
+			LogMessage(fmt.Sprintf(" ..... ERROR: creating new devfile : %v", err))
+		} else {
+			newDevfile.SetSchemaVersion(devfile.SchemaDevFile.SchemaVersion)
+
+			// add the commands to new devfile
+			for _, command := range devfile.SchemaDevFile.Commands {
+				newDevfile.AddCommands(command)
+			}
+			// add components to the new devfile
+			newDevfile.AddComponents(devfile.SchemaDevFile.Components)
+
+			ctx := devfileCtx.NewDevfileCtx(fileName)
+
+			err = ctx.SetAbsPath()
+			if err != nil {
+				LogMessage(fmt.Sprintf(" ..... ERROR: setting devfile path : %v", err))
+			} else {
+				devObj := parser.DevfileObj{
+					Ctx:  ctx,
+					Data: newDevfile,
+				}
+				err = devObj.WriteYamlDevfile()
+				if err != nil {
+					LogMessage(fmt.Sprintf(" ..... ERROR: wriring devfile : %v", err))
```

Comment:
again, I saw the formats for log messages are inconsistent. 
here is ` ..... ERROR: ` for error messages, and above are using ` --> ERROR`
and I saw some messages are with 5 dots, and some are with 6 dots, and with/without spaces in between. 

this is due to the log format is manually input for every message, and which is something we want to avoid

Reply:
Yes, that will help with consistency, part 2? :-)

Reply:
Done